gateway/1.5/modules/ROOT/pages/policies-pdk-create-schema-definition.adoc

```diff
@@ -1,5 +1,362 @@
+        - name: extension-definition
+    properties: 
+        stringProperty: 
+        type: string
+----
+
+The `gcl.yaml` file contains two sections:
+
+* `metadate`: Defines information about the policy
+* `properties`: Defines the policy's configuration parameters
+
+To define a schema definition:
+
+. <<define-metadata>>
+. <<define-parameters>>
+. <<use-custom-parameters-in-rust-source-code>>
+
+== Define Metadata
+
+By adding labels in the metadata section, you can enrich the data shown in the UI when applying the policy.
+
+To define the metadata, edit the `metadata` in the `gcl.yaml` file.
+
+The supported labels are:
+
+* `name`: The policy name that appears in the UI
+* `description`: A brief explanation of the policy's behavior
+* `category`: The category the policy appears under. Supported values are: `security`, `compliance`, `transformation`, `quality of service`, and `troubleshooting`. 
+
+For example:
+
+[source,yaml]
+----
+metadata:
+    labels:
+        title: Json Threat Protection
+        description: Protects against malicious JSON in API requests.
+        category: Security
+----
+
+`labels` are suggested but not required.
+
+== Define Parameters
+
+To add parameters, edit `spec.properties` section  in the `gcl.yaml` file.
+
+For example:
+
+[source,yaml]
+----
+properties:
+    username:
+        type: string
+    password:
+        type: string
+----
+
+After defining your properties, you can:
+
+* <<define-required-properties-and-default-values>>
+* <<define-supported-types>>
+
+=== Define Required Properties and Default Values
+
+For each property in the `gcl.yaml` file, you can define the property as `required` and set a `default` value.
+
+You define properities as required by listing them in the `spec.required` object. For example, the following policy requires a `password` but not `username`:
+
+[source,yaml]
+----
+properties:
+    username:
+        type: string
+    password:
+        type: string
+required:
+    - password
+----
+
+In this configuration example, configuring a policy without a password results in a failed configuration.
+
+You can also define a default value for a property. For Example:
+
+[source,yaml]
+----
+properties:
+    username:
+        type: string
+        default: user1
+    password:
+        type: string
+        default: 12345678
+required:
+    - password
+----
+
+Defining a default value for a required property means that if the field is left blank when configuring the policy, the parameter is set to the default value. Defining a default value for a parameter that isn't required, means that the default value appears as a suggested value in the UI. If a non-required parameter with a default value is left blank during configuration, there is no configured value for the parameters. In the following configuration table, if the policy was applied in API Manager, `user1` would appear as a suggested value for `username` but does not affect if the value is configured:
```

Comment:
```suggestion
If you don't set a value for a required property when configuring a policy, the property uses the default value. The UI suggests the configured default value for properties that are not required. If you don't specify the value for non-required properties, the property does not use the default value. 

In the following configuration table, if the policy was applied in API Manager, `user2` would appear as a suggested value for `username` but does not affect if the value is configured:
```
I tried to make this sentence a bit more clear and less wordy, please check if it is Ok before applying the suggestion. I also changed the value from user1 to user2 to match the example, not sure if that's right.